src/resources/js/tickets.js

```diff
@@ -193,879 +132,962 @@ var ticketHeaderImage = window.ticketHeaderImage || {};
+			provider_id = $checked_provider.val() + '_radio';
+		 }
+
+		$( document.getElementById( provider_id ) ).prop( 'checked', true ).trigger('change');
+	}
+
+	/**
+	 * Returns the current global capacity (via the settings panel.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	function get_global_cap() {
+		var $global_capacity_edit = $( document.getElementById( 'settings_global_capacity_edit' ) )
+		return ( 0 < $global_capacity_edit.length && 0 < $global_capacity_edit.val() ) ? $global_capacity_edit.val() : 0;
+	}
+
+	/**
+	 * When a ticket type is edited we should (re-)establish the UI for showing
+	 * and hiding its history, if it has one.
+	 */
+	function show_hide_ticket_type_history() {
+		var $history = $tribe_tickets.find( '.ticket_advanced.history' );
+
+		if ( ! $history.length ) {
+			return;
 		}
 
-		var datepickerOpts = {
-			dateFormat     : date_format,
-			showAnim       : 'fadeIn',
-			changeMonth    : true,
-			changeYear     : true,
-			numberOfMonths : 3,
-			firstDay       : startofweek,
-			showButtonPanel: false,
-			onChange       : function() {
-			},
-			onSelect       : function( dateText, inst ) {
-				var the_date = $.datepicker.parseDate( 'yy-mm-dd', dateText );
-				if ( inst.id === 'ticket_start_date' ) {
-					$ticket_end_date.datepicker( 'option', 'minDate', the_date );
-				} else {
-					$ticket_start_date.datepicker( 'option', 'maxDate', the_date );
-				}
+		var $toggle_link      = $history.find( 'a.toggle-history' );
+		var $toggle_link_text = $toggle_link.find( 'span' );
+		var $history_list     = $history.find( 'ul' );
+
+		$history.on( 'click', '.toggle-history', function( e ) {
+			e.preventDefault();
+			if ( $history.hasClass( '_show' ) ) {
+				$history.removeClass( '_show' );
+			} else {
+				$history.addClass( '_show' );
 			}
+		} );
+	}
+
+	/**
+	 * Switch from one panel to another
+	 * @param  event e      triggering event
+	 * @param  object ($base_panel) $panel jQuery object containing the panel we want to switch to
+	 * @return void
+	 */
+	function show_panel( e, $panel ) {
+		if ( e ) {
+			e.preventDefault();
+		}
+
+		// this way if we don't pass a panel, it works like a 'reset'
+		if ( undefined == $panel ) {
+			$panel = $base_panel;
+		}
+
+		// First, hide them all!
+		$panels.each( function() {
+			$( this ).attr( 'aria-hidden', true );
+		} );
+
+		// then show the one we want
+		$panel.attr( 'aria-hidden', false );
+
+		if ( ! $panel.is( $base_panel ) ) {
+			$( window ).on( 'beforeunload.tribe', beforeUnload );
+		} else {
+			$( window ).off( 'beforeunload.tribe' );
+		}
+	}
+
+	/**
+	 * Refreshes the base and settings panels when we've changed something
+	 *
+	 * @since TBD
+	 *
+	 * @param string optional notice to prepend to the ticket table
+	 * @param bool (true) flag for panel swap
+	 * @return void
+	 */
+	function refresh_panels( notice, swap ) {
+		// make sure we have this for later (default to true)
+		swap = undefined === swap ? true : false;
```

Comment:
* Should we use `typeof` here to test if it is undefined?
* The ternary will result in true or false ... can't we simplify to just capture the result of the test? (ie, something like `swap = ( 'undefined' === typeof swap );`